azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/auth/base_renewable_token_authentication_provider.py

```diff
@@ -81,14 +96,108 @@ def generate_new_sas_token(self):
             token = _device_token_format.format(quoted_resource_uri, signature, str(expiry))
 
         self.sas_token_str = str(token)
+        self._schedule_token_update(self.token_renewal_interval - self.token_renewal_margin)
+        self._notify_token_updated()
+
+    def set_token_renewal_interval(self, token_renewal_interval, token_renewal_margin):
```

Comment:
>set_token_renewal_interval [](start = 8, length = 26)

not sure how much value this brings compared to editing the two attributes directly? they seem to be public and I seem to recall getters/setters aren't really a thing in python? is the only benefit to have logging?

Reply:
Not much value.  I previously had this function with the side-effect of renewing the token, but I wanted that to be manual.  I'm going to remove this and just do attributes.

---
In reply to: [240404233](https://github.com/Azure/azure-iot-sdk-python-preview/pull/33#discussion_r240404233) [](ancestors = 240404233)